operators/operating-pinot/upsert-merge-compact-task.md

```diff
@@ -0,0 +1,232 @@
+# Upsert Compact Merge Task (Small Segment Merger)
+
+The Upsert Compact Merge Task, also known as the small segment merger, allows you to merge small segments into larger ones for upsert-enabled real-time tables, reducing the total segment count while maintaining storage efficiency.
+
+This task is only supported for REALTIME tables with upsert enabled.
+
+> **ℹ️ New in Pinot 1.3.0**: This task was introduced in version 1.3.0 to address segment proliferation in upsert tables. It works alongside the existing UpsertCompactionTask but focuses on merging multiple small segments rather than compacting individual segments.
+
+> **⚠️ Important Bug Fix**: An issue was identified ([#15846](https://github.com/apache/pinot/issues/15846)) in the initial implementation, which has been fixed via [PR #16034](https://github.com/apache/pinot/pull/16034) and [PR #16086](https://github.com/apache/pinot/pull/16086). These fixes are not present in Pinot 1.3.0. If you plan to use this task, make sure to cherry-pick these PRs or wait for Pinot 1.4.0 release.
+
+## Task overview
+
+The Upsert Compact Merge Task (small segment merger) addresses the challenge of segment proliferation in upsert tables, where the number of segments can grow continuously over time, leading to operational challenges and performance degradation. While individual segments may be efficiently compacted by the UpsertCompactionTask, the overall segment count continues to increase.
+
+### Relationship to UpsertCompactionTask
+
+The Upsert Compact Merge Task complements the existing UpsertCompactionTask:
+- **UpsertCompactionTask**: Compacts individual segments by removing invalid records within each segment
+- **UpsertCompactMergeTask**: Merges multiple small compacted segments into larger ones to reduce total segment count
```

Comment:
it not only merges small compacted segments but also compacts every segment while merging. we can maybe rephrase to make this more clear.